sdk/communication/AzureCommunication/Tests/ObjCCommunicationUserCredentialAsyncTests.m

```diff
@@ -0,0 +1,159 @@
+    XCTestExpectation *expectation = [self expectationWithDescription:
+                                      @"RefreshTokenOnDemand_AsyncRefresh"];
+    __weak ObjCCommunicationUserCredentialAsyncTests *weakSelf = self;
+    
+    CommunicationUserCredential *credential = [[CommunicationUserCredential alloc]
+                                               initWithInitialToken:self.sampleExpiredToken
+                                               refreshProactively:NO
+                                               error:nil
+                                               tokenRefresher:
+                                               ^(void (^ _Nonnull block)
+                                                 (NSString * _Nullable token,
+                                                  NSError * _Nullable error)) {
+            weakSelf.fetchTokenCallCount += 1;
+            block(weakSelf.sampleToken, nil);
+    }];
+    
+    [credential tokenWithCompletionHandler:^(CommunicationAccessToken * _Nullable accessToken,
+                                             NSError * _Nullable error) {
+        XCTAssertNotNil(accessToken);
+        XCTAssertNil(error);
+        XCTAssertEqual(accessToken.token, weakSelf.sampleToken);
+        XCTAssertEqual(accessToken.expiresOn.timeIntervalSince1970, weakSelf.sampleTokenExpiry);
+        
+        [expectation fulfill];
+    }];
+    
+    [self waitForExpectations:@[expectation] timeout:5.0];
+}
+
+- (void)test_ObjCRefreshTokenProactively_TokenExpiringInOneMin {
+    XCTestExpectation *expectation = [self expectationWithDescription:
+                                      @"RefreshTokenProactively_TokenExpiringInOneMin"];
+    __weak ObjCCommunicationUserCredentialAsyncTests *weakSelf = self;
+    
+    NSString *token = [self generateTokenValidForMinutes: 1];
+    CommunicationUserCredential *credential = [[CommunicationUserCredential alloc]
+                                               initWithInitialToken:token
+                                               refreshProactively:YES
+                                               error:nil
+                                               tokenRefresher:
+                                               ^(void (^ block)
+                                                 (NSString * _Nullable newToken,
+                                                  NSError * _Nullable error)) {
+        weakSelf.fetchTokenCallCount += 1;
+        block(weakSelf.sampleToken, nil);
+    }];
+    
+    [credential tokenWithCompletionHandler:^(CommunicationAccessToken * _Nullable accessToken,
+                                             NSError * _Nullable error) {
+        XCTAssertNotNil(accessToken);
+        XCTAssertEqual(accessToken.token, weakSelf.sampleToken);
+        XCTAssertEqual(weakSelf.fetchTokenCallCount, 1);
+        
+        [expectation fulfill];
+    }];
+
+    [self waitForExpectations:@[expectation] timeout:2.0];
```

Comment:
why some test cases use 5 and others use 2 for timeout values?

Reply:
We can change them all to 2 I was previously following the the swift tests. But I noticed that some async calls were doing dispatch after. Since I removed the that little logic we can probably just have all the time outs be 2 seconds